infor-design/enterprise-wc
# 4972 - ids rating wc

**Explain the details for making this change. What existing problem does the pull request solve?**
Porting and converting the enterprise jQuery rating component into an IDS Web Component. Creating a custom element <ids-rating> web component with options to customize the stars, readonly, and full/half values. When writable as the stars are clicked the value setAttribute will change to reflect the updated star count. 

**Related github/jira issue (required)**:
Closes https://github.com/infor-design/enterprise/issues/4972

**Steps necessary to review your pull request (required)**:
- Checkout the feature branch - _**git checkout 4972-ids-rating-wc**_
- Pull the latest changes - _**git pull -f**_
- Create a sample html document inside of the ids-rating folder - _**./app/ids-rating**_
- Create the custom element by using the markup - _**`<ids-rating></ids-rating>`**_
- Navigate to the ratings wc url - _**http://localhost:4300/ids-rating**_
- Click on the given stars and check for the update value in the web inspect.
<img width="1262" alt="Screen Shot 2021-04-28 at 8 59 22 AM" src="https://user-images.githubusercontent.com/5215954/116407811-25a57980-a800-11eb-84d1-1d197eb7c189.png">
<img width="764" alt="Screen Shot 2021-04-28 at 9 00 34 AM" src="https://user-images.githubusercontent.com/5215954/116407895-39e97680-a800-11eb-8371-5832c5804641.png">
<img width="628" alt="Screen Shot 2021-04-28 at 9 01 36 AM" src="https://user-images.githubusercontent.com/5215954/116408135-761cd700-a800-11eb-94a1-2e343958a6b1.png">

**Pending completion in this Pull Request**:
- e2e fn testing - current coverage 26%
- Percy testing
- ReadMe Docs

**Included in this Pull Request**:
